jumpscale/sals/reservation_chatflow/deployer.py

```diff
@@ -63,8 +64,59 @@ def add_node(self, node, pool_id):
             j.sals.zos.network.add_node(network, node.node_id, str(subnet), pool_id)
             return network
 
-    def add_access(self, node, pool_id):
-        pass
+    def add_access(self, node_id=None, use_ipv4=True):
+        node_id = node_id or self.network_workloads[0].info.node_id
+        used_ip_ranges = set()
+        for workload in self.network_workloads:
+            used_ip_ranges.add(workload.iprange)
+            for peer in workload.peers:
+                used_ip_ranges.add(peer.iprange)
+        else:
+            network_range = netaddr.IPNetwork(self.iprange)
+            for idx, subnet in enumerate(network_range.subnet(24)):
+                if str(subnet) not in used_ip_ranges:
+                    break
+            else:
+                raise StopChatFlow("Failed to find free network")
+        network = j.sals.zos.network.create(self.iprange, self.name)
+        node_workloads = {}
+        for net_workload in self.network_workloads:
+            node_workloads[net_workload.info.node_id] = net_workload
+        network.network_resources = list(node_workloads.values())  # add only latest network resource for each node
+        wg_quick = j.sals.zos.network.add_access(network, node_id, str(subnet), ipv4=use_ipv4)
+        return network, wg_quick
+
+    def delete_access(self, ip_range, node_id=None):
```

Comment:
This logic seems like it should belong to the network SAL itself instead of living in the chat flow code.

Reply:
move to sal
https://github.com/threefoldtech/js-sdk/pull/260/commits/60103f27345df6898d4cdd3b7a3549f666ed8117#diff-2df053358d67fef7cac6080cd667c92eR183